corelog/formatters.go

```diff
@@ -0,0 +1,103 @@
+package corelog
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+)
+
+var pid = os.Getpid()
+
+type Formatter interface {
+	Format(pkg string, level LogLevel, depth int, entries ...LogEntry)
+	SetWriter(w io.Writer)
+}
+
+type StringFormatter struct {
+	w *bufio.Writer
+}
+
+func (s *StringFormatter) Format(pkg string, _ LogLevel, _ int, entries ...LogEntry) {
+	s.w.WriteString(pkg)
+	endsInNL := false
+	for _, v := range entries {
+		s.w.WriteByte(' ')
+		str := v.LogString()
+		endsInNL = strings.HasSuffix(str, "\n")
+		s.w.WriteString(str)
+	}
+	if !endsInNL {
+		s.w.WriteString("\n")
+	}
+	s.w.Flush()
+}
+
+func (s *StringFormatter) SetWriter(w io.Writer) {
+	s.w = bufio.NewWriter(w)
+}
+
+type GlogFormatter struct {
+	StringFormatter
+}
+
+func (g GlogFormatter) Format(pkg string, level LogLevel, depth int, entries ...LogEntry) {
+	g.w.Write(GlogHeader(level, depth+1))
+	g.StringFormatter.Format(pkg, level, depth, entries...)
+}
+
+func GlogHeader(level LogLevel, depth int) []byte {
+	// Lmmdd hh:mm:ss.uuuuuu threadid file:line]
+	now := time.Now()
```

Comment:
Then just use StringFormatter? Or help me define a default formatter that doesn't have a timestamp. It shouldn't be an option, it should be a (built-in) Formatter